assets/js/controllers/AppCtrl.coffee

```diff
@@ -132,40 +134,41 @@ app.controller 'AppCtrl', [
     $scope.$on 'ws/dataset', (event, payload) ->
       backendService.getExperiment()
         .then (experiment) ->
-          dataset = experiment.dataset
-          if payload.data.status == 'done' and payload.pk == dataset.id
+          datasetId = experiment.datasetId
+          if payload.data.status == 'done' and payload.pk == datasetId
             $scope.reloadDataset()
         .fail console.error
 
-    $scope.$watch 'dataset', ((newValue, oldValue) ->
+    $scope.$watch 'datasetId', ((newValue, oldValue) ->
       if newValue?
         $scope.reloadDataset()
         systemStatus.waitForDatasetProcessed()
       return
       ), true
 
     $scope.$watchCollection 'selectedFeatures', (newSelectedFeatures) ->
+      return unless $scope.featuresLoaded
+      newSelectedFeatureIds = newSelectedFeatures.map (f) -> f.id
       backendService.getExperiment()
-        .then (experiment) -> experiment.setSelection newSelectedFeatures
+        .then (experiment) -> experiment.setSelection newSelectedFeatureIds
         .fail console.error
 
     $scope.initializeFromExperiment = (experiment) ->
       $scope.targetFeatureId = experiment.targetId
-      $scope.dataset = {id: experiment.dataset.id, name: experiment.dataset.name}
-      selection = experiment.getSelection()
-      if selection?
-        $scope.selectedFeatures = selection
```

Comment:
Does the experiment no longer contain the selection? I'm wondering since $scope.selectedFeatures isn't set to anything from the experiment in `resetExperimentData()` either.